### PR TITLE
Game template code fix

### PR DESCRIPTION
**Pull request by benanil**
android code was dependent to game RenderTarget android code is writing game rendertarget to back buffer
this pr includes web build fix.

### CODE REVIEW

File: Template/Codes/Game.cpp [hunk 1]
```diff
@@ -26,31 +26,37 @@
...
+  void Game::Init(Main* master)
+  { 
+    m_sceneRenderer = MakeNewPtr<SceneRenderer>();
+    ScenePtr scene  = GetSceneManager()->Create<Scene>(ScenePath("Engine/Scenes/ms-ball.scene"));
+    m_sceneRenderer->m_params.Scene = scene;
+  }
 
-  void Game::Init(Main* master) { Main::SetProxy(master); }
-
-  void Game::Destroy() { delete this; }
```
Comment (Osman-Fatih-Cakir) [inline Template/Codes/Game.cpp:36]: game should destroy itself

File: Editor/LightMeshGenerator.cpp [hunk 2]
```diff
@@ -103,77 +100,47 @@ namespace ToolKit
...
-      }
-      if (d.y == 0.0f)
-      {
-        zeroCount++;
-      }
-      if (d.z == 0.0f)
-      {
-        zeroCount++;
-      }
+      int zeroCount = 0;
```
Comment (afraidofdark) [inline Editor/LightMeshGenerator.cpp]: how come this old changes appear on top of this PR ?

Comment (benanil) [reply Editor/LightMeshGenerator.cpp]: I don't know why this happens
I have done this commands
git checkout Engine
git revert HEAD
git reset --hard
git pull
git stash pop
git commit 
git push

Comment (afraidofdark) [reply Editor/LightMeshGenerator.cpp]: after checkout the Engine branch why do you Revert ?
when you are about to stasrt a new branch always make sure you are at the "Engine" branch.
git chekcout Engine
Make sure Engine is clean and uptodate
git reset --hard
git pull
than create your task branch 
git checkout -b dev/anl/task-id

**Pull request closed without merging**